---
title: "Advanced Genomics project 3"
author: "Gabriele Gambero"
date: "2024-03-19"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Single cells, transcription factor factor activity proxy, and
post-translational modifications. The activity of a transcription factor is in principle
detectable by observing the expression level of its targets. Indeed, we usually
assume that genes expressed similarly are often controlled by a same regulator, and
therefore also the reverse should be true. 
In this project, we want to introduce a proxy for a transcription factor activity based on the expression level of genes
regulated by that transcription factor. The regulator-target relationships can be found
in specialized databases, such as regulondb for coli, yeastract for yeast... For a
positive regulator, its activity can be taken as being a function (linear? sigmoidal?) of
the expression level of its targets (mean? median? max?). Now, the question is to
show the degree at which transcription factor activities are correlated to the
corresponding transcription factor gene expression level. Our null hypothesis is that
indeed, the abundance of the transcript is also a proxy of the activity. This can be
somehow expected, but plausibly not for all transcription factors. Many for instance
are active only in phosphorylated form, meaning that the transcript can also be
constitutively expressed; the activity will depend on phosphorylation however, and
therefore it will be correlated to the activity of the corresponding kinase. In this way,
one could derive a general model to relate the activity of a TF and its transcriptâ€™s
abundance, and in case the correlation is significant, to identify those transcription
factors significantly deviating from the general behavior.




```{r}
regul <- read.table("TF-RISet.tsv", sep = '\t', header = T)
dim(regul)
head(regul)
```
1)riId. Regulatory interaction (RI) identifier assigned by RegulonDB
2)riType. Regulatory interaction type [tf-promoter, tf-tu, tf-gene,srna-promoter, srna-tu, srna-gene, compound-promoter, compound-tu, compound-gene]
3)regulatorId. Transcription Factor (TF) identifier assigned by RegulonDB
4)regulatorName. Regulator name
5)cnfName. regulator active conformation name
6)tfrsID. regulator regulatory site (TFRS) identifier assigned by RegulonDB
7)tfrsLeft. TFRS left end position in the genome
8)tfrsRight. TFRS right end position in the genome
9)strand. DNA strand where the TFRS is located
10)tfrsSeq. TFRS sequence (upper case)
11)riFunction. Gene expression effect caused by the TF bound to the TFRS
12)promoterID. Promoter Identifier assigned by RegulonDB
13)promoterName. Promoter name
14)tss. Transcription start site (+1) position in the genome
15)sigmaF. Sigma Factor that recognize the promoter
16)tfrsDistToPm. Relative distance from the center position of TFRS to the Transcription Start Site
17)firstGene. first transcribed gene name
18)tfrsDistTo1Gene. Relative distance from center position of TFRS to the start of first gene
19)targetTuOrGene. Transcription unit or gene (id:name) regulated by the TF
20)confidenceLevel. RI confidence level (Values: Confirmed, Strong, Weak)
21)tfrsEvidence. Evidence that supports the existence of the TFRS [EvidenceCode|EvidenceType(C:confirmed S:strong W:weak)]]
22)riEvidence. Evidence that supports the RI function [EvidenceCode|EvidenceType(C:confirmed S:strong W:weak)]
23)addEvidence. Additive Evidence [CV(EvidenceCode1/EvidenceCodeN)|Confidence Level]
24)riEvTech. Evidence related to the type of technology used to determine the RI
25)riEvCategory . Evidence were categorized in classical, ht or non-experimental





```{r}
expr <- read.table("./SBRG/data/log_tpm_norm.csv", sep = ",", header = T)
head(expr)
dim(expr)
```
```{r}
expr[1000,]
```


All genes names:



```{r}
names <- read.table("./SBRG/data/imodulon_gene_names.txt", sep = ",", header = T)
head(expr)
dim(expr)
```

```{python}
names = []

with open('./SBRG/data/imodulon_gene_bnumbers.txt', 'r') as all_b:
  scartare = all_b.readline()
  for line in all_b.readlines():
    line_genes = line.split(',')
    for g in line_genes:
      names.append(g)
      

print(len(names))
names[0:10]
```

```{python}
# ,start,stop,strand,gene_name,length,operon,cog
import pandas as pd

with open("./SBRG/data/log_tpm_norm.csv", 'r') as to_sub:
  scartare2 = to_sub.readline()
      
  for riga in to_sub.readlines():
    riga_db = riga.split(',')

    with open('./SBRG/data/gene_info.csv', 'r') as all_b:
      scartare = all_b.readline()
          
      for line in all_b.readlines():
        line_genes = line.split(',')
    
        if line_genes[0] == riga_db[0]
            
            
    for g in line_genes:
      names.append(g)

```

```{python}
import pandas as pd

to_sub = pd.read_csv("./SBRG/data/log_tpm_norm.csv",)
to_sub.iloc[0:8,0:5]

```















































